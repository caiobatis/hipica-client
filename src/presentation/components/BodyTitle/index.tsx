import { Grid, Grid2, Typography } from '@mui/material'

type Props = {
  number: number
  title: string
  description: string
}

export function BodyTitle({ number, title, description }: Props) {
  return (
    <Grid xs={1} alignItems="center" display="flex" item py={2}>
      {/* <Box
        width={40}
        height={40}
        minWidth={40}
        borderRadius={20}
        bgcolor="black"
        display="flex"
        alignItems="center"
        justifyContent="center"
        mr={2}
      >
        <Typography variant="h3" component="div" color="white">
          {number}
        </Typography>
      </Box> */}

      <Grid2>
        <Typography variant="body2">Ol√° diretor(a),</Typography>
        <Typography variant="h4" component="div">
          {title}
        </Typography>

        {/* <Typography variant="body1" maxWidth={700}>
          {description}
        </Typography> */}
      </Grid2>
    </Grid>
  )
}
