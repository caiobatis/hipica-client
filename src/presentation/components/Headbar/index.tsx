import { Button, Grid2, Typography } from '@mui/material'
import { useNavigate } from 'react-router-dom'

export function Headbar({ hasBackButton = true }) {
  const navigate = useNavigate()
  return (
    <div>
      <Grid2
        container
        mt={2}
        alignItems="center"
        justifyContent="space-between"
      >
        <Grid2 mb={2}>
          <Typography variant="h4" mb={1}>
            Ajude a organizar nossos cultos
          </Typography>

          <Typography variant="body1" mb={1}>
            Organize sua escala e contribua com a organizacao dos cultos.
          </Typography>
        </Grid2>

        {hasBackButton && (
          <Grid2>
            <Button
              type="button"
              variant="outlined"
              size="large"
              fullWidth
              onClick={() => navigate(-1)}
            >
              Voltar
            </Button>
          </Grid2>
        )}
      </Grid2>
    </div>
  )
}
