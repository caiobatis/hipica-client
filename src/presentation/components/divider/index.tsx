export function Divider() {
  return (
    <div role="separator">
      <div>|</div>
    </div>
  )
}
