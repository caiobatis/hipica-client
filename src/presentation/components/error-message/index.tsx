import { testProps } from '~/tests/testProps'
import { ButtonTryAgain, Container, Subtitle, Title } from './styles'
import type { ErrorMessageProps } from './types'

export function ErrorMessage({
  title,
  image = 'emoji-warning',
  subtitle,
  onTryAgain,
}: ErrorMessageProps) {
  return (
    <Container>
      <img src={`/assets/images/${image}.svg`} alt={image} />
      <Title variant="heading-medium">{title}</Title>
      <Subtitle variant="text-medium" color="medium">
        {subtitle}
      </Subtitle>
      {onTryAgain && (
        <ButtonTryAgain
          variant="primary-solid"
          onClick={onTryAgain}
          {...testProps('btn-try-again')}
        >
          Tentar novamente
        </ButtonTryAgain>
      )}
    </Container>
  )
}
