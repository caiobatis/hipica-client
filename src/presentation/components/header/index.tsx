import ArrowBackIcon from '@mui/icons-material/ArrowBack'
import ExitToAppIcon from '@mui/icons-material/ExitToApp'
import GradingIcon from '@mui/icons-material/Grading'
import MenuIcon from '@mui/icons-material/Menu'
import PeopleIcon from '@mui/icons-material/People'
import {
  Box,
  Container,
  Divider,
  Drawer,
  Grid,
  IconButton,
  List,
  ListItem,
  ListItemButton,
  ListItemIcon,
  ListItemText,
  Typography,
} from '@mui/material'
import { useState } from 'react'
import { Link, useNavigate } from 'react-router-dom'

type Props = {
  email?: string
}

export function Header({ email }: Props) {
  const navigate = useNavigate()

  const logout = () => {
    localStorage.removeItem('email')
    localStorage.removeItem('date_')

    document.location.reload()
  }

  const [open, setOpen] = useState(false)

  const toggleDrawer = (newOpen: boolean) => () => {
    setOpen(newOpen)
  }

  const DrawerList = (
    <Box sx={{ width: 250 }} role="presentation" onClick={toggleDrawer(false)}>
      <List>
        <ListItem disablePadding>
          <ListItemButton>
            <ListItemText primary="IASD Portal da Hípica" />
          </ListItemButton>
        </ListItem>
      </List>
      <Divider />
      <List>
        <Link to="/departamento">
          <ListItem disablePadding>
            <ListItemButton>
              <ListItemIcon sx={{ minWidth: 40 }}>
                <PeopleIcon sx={{ color: 'black' }} />
              </ListItemIcon>

              <Typography variant="body1" color="black">
                Gerir de escalas
              </Typography>
            </ListItemButton>
          </ListItem>
        </Link>

        <Link to="/escala-geral">
          <ListItem disablePadding>
            <ListItemButton>
              <ListItemIcon sx={{ minWidth: 40 }}>
                <GradingIcon sx={{ color: 'black' }} />
              </ListItemIcon>

              <Typography variant="body1" color="black">
                Visualizar escalas
              </Typography>
            </ListItemButton>
          </ListItem>
        </Link>
      </List>
    </Box>
  )

  return (
    <>
      <Box sx={{ py: 2, bgcolor: '#625784' }}>
        <Container fixed>
          <Grid container columns={{ xs: 4 }}>
            <Grid item display="flex" xs={3}>
              {['/departamento', '/escala-geral'].includes(
                window.location.pathname,
              ) ? (
                <IconButton
                  color="info"
                  sx={{ bgcolor: 'transparent !important', color: 'white' }}
                  onClick={toggleDrawer(true)}
                >
                  <MenuIcon />
                </IconButton>
              ) : (
                <IconButton
                  color="info"
                  sx={{ bgcolor: 'transparent !important', color: 'white' }}
                  onClick={() => navigate(-1)}
                >
                  <ArrowBackIcon />
                </IconButton>
              )}

              <Box ml={1}>
                <Typography
                  variant="body1"
                  fontWeight={600}
                  lineHeight={1.5}
                  color="white"
                >
                  {window.location.pathname === '/departamento'
                    ? 'Gestão de escalas'
                    : 'Escala geral'}
                </Typography>
                <Typography variant="body2" color="white">
                  {email}
                </Typography>
              </Box>
            </Grid>

            <Grid item xs={1} display="flex" justifyContent="flex-end">
              <IconButton
                color="info"
                sx={{ bgcolor: 'transparent !important', color: 'white' }}
                onClick={logout}
              >
                <ExitToAppIcon />
              </IconButton>
            </Grid>
          </Grid>
        </Container>
      </Box>

      <Drawer open={open} onClose={toggleDrawer(false)}>
        {DrawerList}
      </Drawer>
    </>
  )
}
