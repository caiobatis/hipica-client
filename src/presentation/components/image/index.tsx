import { useCallback, useEffect, useState } from 'react'
import { testProps } from '~/tests/testProps'
import { FallBack, ImageStyled } from './styles'
import type { ImageProps } from './types'

/**
 * The props for the Image component.
 * @param width the width of the image.
 * @param height the height of the image.
 * @param source the source of the image.
 * @param fit the fit of the image. Can be cover or contain.
 * @param borderRadius the border radius of the image. Can be container or interactive. Container is 8px and interactive is 4px.
 * @param rest the rest of the props for the image. Ex.: data-testid, className, etc.
 * @returns an image component.
 * @example
 * <Image width="100px" height="100px" source="https://example.com/image.png" />
 */

export function Image({
  width,
  source,
  height,
  fit,
  borderRadius,
  ...props
}: ImageProps) {
  const [error, setError] = useState(false)

  useEffect(() => {
    setError(false)
  }, [source])

  const onError = useCallback(() => {
    setError(true)
  }, [])

  return (
    <>
      {error || !source ? (
        <FallBack
          width={width}
          height={height}
          {...testProps('component-fallback')}
        >
          <p>
            {width} x {height}
          </p>
        </FallBack>
      ) : (
        <ImageStyled
          width={width}
          height={height}
          src={source}
          onError={onError}
          role="img"
          $fit={fit}
          $borderRadius={borderRadius}
          {...props}
        />
      )}
    </>
  )
}

export default Image
