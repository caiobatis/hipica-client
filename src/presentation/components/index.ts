export * from './divider'
export * from './error-message'
export * from './image'
export * from './page-loader'
export * from './page-title'
export * from './sidebar'
