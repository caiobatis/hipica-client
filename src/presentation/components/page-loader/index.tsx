import { testProps } from '~/tests/testProps'
import { Container, Loading } from './styles'

export function PageLoader() {
  return (
    <Container {...testProps('page-loader')}>
      <Loading>loading</Loading>
    </Container>
  )
}
