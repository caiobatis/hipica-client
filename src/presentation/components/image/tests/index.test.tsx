import { afterAll, describe, expect, it } from 'vitest'
import { mockJadeComponents } from '~/infra/mocks'
import { Image } from '~/presentation/components'
import type { ImageProps } from '~/presentation/components/image/types'
import { fireEvent, render, type RenderResult } from '~/tests'
import { testSnapshots } from '~/tests/utils'
import * as states from './componentStates'

mockJadeComponents()

const defaultProps: ImageProps = {
  alt: 'alt',
  height: 100,
  source: 'ton-logo.svg',
  width: 100,
}

const makeSut = (props: Partial<ImageProps> = {}): RenderResult => {
  return render(<Image {...defaultProps} {...props} />)
}

describe('Image', (): void => {
  testSnapshots(states)

  let component: RenderResult

  afterAll(() => {
    component.unmount()
  })

  it('should render image', () => {
    component = makeSut({ source: '123', title: 'image' })

    const image = component.queryByTitle('image')

    expect(image).toBeInTheDocument()
  })

  it('should render without source', () => {
    component = makeSut({ source: undefined })

    const fallback = component.queryByTestId('component-fallback')

    expect(fallback).toBeInTheDocument()
  })

  it('should render with borderRadius container', () => {
    component = makeSut({
      borderRadius: 'container',
      source: '123',
      title: 'image',
    })

    const image = component.getByRole('img')

    expect(image).toBeInTheDocument()
  })

  it('should render with onError action', () => {
    component = makeSut({ source: '123', title: 'image' })

    const image = component.getByTitle('image')

    // Fakes an error event
    fireEvent.error(image)

    expect(image).not.toBeInTheDocument()
  })
})
