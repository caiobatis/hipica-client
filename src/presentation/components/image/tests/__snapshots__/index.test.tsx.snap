// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Image > renders 'default' state correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <img
        alt="alt"
        class="jzlLpA"
        height="100"
        role="img"
        src="/assets/images/ton-logo.svg"
        width="100"
      />
    </div>
  </body>,
  "container": <div>
    <img
      alt="alt"
      class="jzlLpA"
      height="100"
      role="img"
      src="/assets/images/ton-logo.svg"
      width="100"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Image > renders 'without source' state correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="hCOkuJ"
        data-testid="component-fallback"
        height="100"
        width="100"
      >
        <span>
          100
           x 
          100
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="hCOkuJ"
      data-testid="component-fallback"
      height="100"
      width="100"
    >
      <span>
        100
         x 
        100
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
